src/lib/promptToEdit.tsx

```diff
@@ -244,55 +274,71 @@ export async function getPromptToEditResult(
 export async function doPromptEdit({
   prompt,
   selections,
-  code,
+  projectFiles,
   token,
   artifactGraph,
   projectName,
 }: {
   prompt: string
   selections: Selections
-  code: string
+  projectFiles: FileMeta[]
   token?: string
   projectName: string
   artifactGraph: ArtifactGraph
-}): Promise<Models['TextToCadIteration_type'] | Error> {
+}): Promise<Models['TextToCadMultiFileIteration_type'] | Error> {
   const toastId = toast.loading('Submitting to Text-to-CAD API...')
-  const submitResult = await submitPromptToEditToQueue({
-    prompt,
-    selections,
-    code,
-    token,
-    artifactGraph,
-    projectName,
-  })
-  if (err(submitResult)) return submitResult
 
-  const textToCadComplete = new Promise<Models['TextToCadIteration_type']>(
-    (resolve, reject) => {
-      ;(async () => {
-        const MAX_CHECK_TIMEOUT = 3 * 60_000
-        const CHECK_DELAY = 200
+  let submitResult
 
-        let timeElapsed = 0
+  // work around for @kittycad/lib not really being built for the browser
+  ;(window as any).process = {
+    env: {
+      ZOO_API_TOKEN: token,
+      ZOO_HOST: VITE_KC_API_BASE_URL,
+    },
+  }
```

Comment:
Obviously a hack, and naturally one asks, why not fix @kittcad/lib to make it more browser friendly to not _NEED_ env vars like this?

And my thinking was the @kittycad/lib should not be browser friendly because anyone integrating with our product and using @kittycad/lib should proxy their requests on their on backend because they should not be putting their integration API key into the browser.

The only reason why it's okay for us, is because the API key is the user's API key, so they can't steal something that's already theirs. So I opted to leave the library as is and keep this work around here.

Reply:
Maybe as a follow up you could add some of this context to the comment block? It would help our team and outside developers learning from our source

Reply:
Also if you add `process` to [this type declaration file](https://github.com/KittyCAD/modeling-app/blob/kurt-multi-file-edit-with-ai/interface.d.ts#L117-L120) you won't need to do this `window as any` thing